"Report issue" button

bindetad opened the pull request:
## Info:
* Implements a "Report Issue" button as specified.
* Adds Czech language properly.
* Can be merged right away.
## Deploy previews:
[ML Test](https://deploy-preview-227--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-1101091271)
[Datové fondy VS](https://deploy-preview-227--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-1153182747)
[Marie test](https://deploy-preview-227--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-530560086)
[mm-test](https://deploy-preview-227--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-2000996011)
[Obchodní rejstřík](https://deploy-preview-227--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance201242184)
[Sčítání lidu](https://deploy-preview-227--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-2011762727)
[OFN číselníky a klasifikace](https://deploy-preview-227--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance-1810443144)
[Sbírka](https://deploy-preview-227--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance1483547911)
[časoprostor](https://deploy-preview-227--ontographer.netlify.app/?workspace=https://slovn%C3%ADk.gov.cz/datov%C3%BD/pracovn%C3%AD-prostor/pojem/metadatov%C3%BD-kontext/instance926567775)

karelklima commented on src/panels/menu/help/HelpCheatSheet.tsx:
```diff
@@ -1,23 +1,24 @@
+			<h4>{LocaleHelp[ProjectSettings.viewLanguage].cheatSheet.title}</h4>
+			<p>{LocaleHelp[ProjectSettings.viewLanguage].cheatSheet.t1}</p>
+			{Object.keys(LocaleHelp[ProjectSettings.viewLanguage].cheatSheet.interactions).map((interaction: string) =>
+				<div key={key++}><h5 key={key++}>{interaction}</h5>
```
It would be better to use `index` for the key, which you can get as a second argument of the mapping function, i.e.:
```
Array.map((interaction: string, index: number) => <div key={index}> ...
```
Because when it is like this, you mutate a variable which is outside of the mapping function, which works but in general it's best to avoid similar side effects / mutate stuff.

bindetad replied:
Implemented.

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"ontoGrapher": "ontoGrapher",
+	"errorMissingRestriction": "Restriction not found",
+	"validationNumber": "#",
+	"deletedRelationships": "Některé vztahy jsou v Compact pohledu skryty. Přepněte zpět na Full pohled pro jejich obnovení.",
```
... v kompaktním pohledu

psiotwo replied:
... úplný pohled

karelklima commented on src/panels/menu/help/HelpCheatSheet.tsx:
```diff
@@ -1,23 +1,24 @@
 import React from 'react';
 import {ProjectSettings} from "../../../config/Variables";
-import {helpModal} from "../../../locale/help";
+import {LocaleHelp} from "../../../config/Locale";
```
You know, if you add the following to `tsconfig.json` `compilerOptions` key:
```
"include": [
  "src"
],
```
You would be able to do this instead:
```
import {LocalHelp} from 'config/Locale'
```
Which helps with refactoring a lot :-)

bindetad replied:
Hmm... I already have this key inserted, but it does not seem to work. If I am completely honest, though, I don't pay attention to imports even when heavily refactoring something, since Webstorm has nice functions that does everything for me, and I never had a problem with it. So I will try to look into it, but if you don't mind, I currently give it a low priority - but thank you nevertheless.

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"brokenConnection": "Vaše připojení k databázi bylo přerušeno.",
+	"description": "Popis",
+	"vocabularyNotFound": "URL definice slovníků chybí nebo není správné",
+	"validate": "Validovat",
```
Zkontrolovat

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"conforms": "Pracovní prostor je podle serveru validní",
+	"validationError": "Zpráva",
+	"validationReload": "Obnovit",
+	"validation": "Validace",
```
Výsledky kontroly

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"validationNumber": "#",
+	"deletedRelationships": "Některé vztahy jsou v Compact pohledu skryty. Přepněte zpět na Full pohled pro jejich obnovení.",
+	"validationName": "Popis",
+	"validationSeverity": "Vážnost",
```
Závažnost

karelklima commented on src/panels/menu/help/HelpMain.tsx:
```diff
@@ -1,16 +1,10 @@
-			<p>{helpModal[ProjectSettings.viewLanguage].intro.t2}</p>
-			<p><a href="https://github.com/opendata-mvcr/ontoGrapher/issues/new"
-				  rel="noopener noreferrer"
-				  target="_blank">https://github.com/opendata-mvcr/ontoGrapher/issues/new</a></p>
```
TBH it would be best if you kept the link to issues as an `a` element in the new solution, because of two reasons:
1) Users see the `href` in the status bar in the browser
2) Browsers very often block window.open() because it's used by ads etc.

But it's not a big deal. I guess it was impossible to implement with the split button requirement.

bindetad replied:
I managed to refactor the SplitButton component such that it is implemented as a <a> with href (a feature of React-Bootstrap).

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"errorConnectionLost": "Chyba: připojení k pracovnímu prostoru bylo přerušeno.",
+	"diagramTab": "Diagramy",
+	"modalNewElemExistsError": "Koncept s tímto jménem již ve vybraném slovníku existuje.",
+	"addAltLabelPlaceholder": "Nový synonym...",
```
Nové synonymum

karelklima commented on src/scss/panels/MenuPanel.scss:
```diff
@@ -23,6 +23,37 @@
+      margin-bottom: 4px;
+
+      &:hover {
+        cursor: default;
```
We need to talk about this some time :-) It's not very user friendly to prevent the pointer cursor to be displayed on a clickable element, but I am aware that it's consistent with the rest of the app.

karelklima reviewed (approved):
Minor issues, left some comments, but no blockers.

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"errorMissingRestriction": "Restriction not found",
+	"validationNumber": "#",
+	"deletedRelationships": "Některé vztahy jsou v Compact pohledu skryty. Přepněte zpět na Full pohled pro jejich obnovení.",
+	"validationName": "Popis",
```
'pojem'

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"detailPanelPrefLabel": "Jméno",
+	"detailPanelAltLabel": "Synonyma",
+	"detailPanelInScheme": "Ve slovníku",
+	"detailPanelDefinition": "Popis",
```
Definice

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"modalNewElemDescription": "Jméno pro nový koncept (nelze později změnit!):",
+	"modalNewElemTitle": "Vytvořit nový koncept",
+	"modalNewLinkTitle": "Vytvořit nový vztah",
+	"changelog": "Changelog",
```
Changelog a jeho obsah je jen anglicky - působí to v českém lokalizaci divně. (ale není to velký problém).

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"view": "Vzhled",
+	"switchColors": "Barvy slovníků",
+	"reportIssue": "Nahlásit chybu",
+	"reportEnhancement": "Navrhnout úpravu",
```
text is OK, but the button looks differently (e.g. it has a different color) than the "nahlásit chybu" button

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"fileProjectSettingsDescriptions": "Popis",
+	"help": "Nápověda",
+	"modalNewElemError": "Jméno konceptu je povinné.",
+	"selectPackage": "Koncept se vytvoří v:",
```
TermIt uses the word "Pojem" in Czech. Although both are defendable, "Pojem" looks more "friendly" than "koncept". I propose to change the wording of all occurrences of "Koncept" to "Pojem"

psiotwo replied:
Koncept se vytvoří ve slovníku

(the combobox items wording is in nominativ)

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"modalNewElemTitle": "Vytvořit nový koncept",
+	"modalNewLinkTitle": "Vytvořit nový vztah",
+	"changelog": "Changelog",
+	"modalNewLinkDescription": "Vyberte vztah z následujících možností:",
```
Vyberte typ vztahu:

psiotwo commented on src/locale/cs.ts:
```diff
@@ -0,0 +1,94 @@
+	"modalRemovePackageItemTitle": "Smazat koncept?",
+	"modalRemoveDiagramTitle": "Smazat diagram?",
+	"modalRemoveDiagramDescription": "Opravdu chcete smazat diagram?",
+	"connectionVia": "přes vztah",
```
typ vztahu

psiotwo reviewed (changes requested):
see comments

bindetad commented on src/config/Locale.ts:
```diff
@@ -1,3 +1,14 @@
+import {csHelp} from "../locale/cshelp";
+
+export const Locale: { [key: string]: { [key: string]: string } } = {
+	en: en,
```
Implemented, thank you.

bindetad commented on src/panels/menu/MenuPanelReport.tsx:
```diff
@@ -0,0 +1,39 @@
+	}
+
+	openReportURL(issue: boolean) {
+		window.open(`https://github.com/opendata-mvcr/sgov-assembly-line/issues/new${
```
Implemented.

bindetad commented on src/panels/menu/MenuPanelReport.tsx:
```diff
@@ -0,0 +1,39 @@
+		</span>);
+	}
+
+	openReportURL(issue: boolean) {
```
I eventually removed this function (more on this in another comment).

bindetad commented:
In 4201b41, I changed the localization as directed, changed the styling of the split button dropdown item to also be yellow, and localized the changelog as well.

psiotwo reviewed (approved):
lgtm
